src/testing/core/__tests__/createTestSchema.test.tsx

```diff
@@ -1028,4 +1028,192 @@ describe("schema proxy", () => {
 
     unmount();
   });
+
+  it("createSchemaFetch respects min and max delay", async () => {
+    const Profiler = createDefaultProfiler<ViewerQueryData>();
+
+    const maxDelay = 2000;
+
+    using _fetch = createSchemaFetch(schema, {
+      delay: { min: 10, max: maxDelay },
+    });
+
+    const client = new ApolloClient({
+      cache: new InMemoryCache(),
+      uri,
+    });
+
+    const query: TypedDocumentNode<ViewerQueryData> = gql`
+      query {
+        viewer {
+          id
+          name
+          age
+          book {
+            id
+            title
+            publishedAt
+          }
+        }
+      }
+    `;
+
+    const Fallback = () => {
+      useTrackRenders();
+      return <div>Loading...</div>;
+    };
+
+    const App = () => {
+      return (
+        <React.Suspense fallback={<Fallback />}>
+          <Child />
+        </React.Suspense>
+      );
+    };
+
+    const Child = () => {
+      const result = useSuspenseQuery(query);
+
+      useTrackRenders();
+
+      Profiler.mergeSnapshot({
+        result,
+      } as Partial<{}>);
+
+      return <div>Hello</div>;
+    };
+
+    const { unmount, rerender } = renderWithClient(<App />, {
+      client,
+      wrapper: Profiler,
+    });
+
+    // initial suspended render
+    await Profiler.takeRender();
+
+    {
+      try {
+        const { snapshot: _snapshot } = await Profiler.takeRender();
+      } catch (e) {
+        // default timeout is 1000, so this throws
+        if (e instanceof Error) {
+          expect(e.message).toMatch(
+            /Exceeded timeout waiting for next render./
+          );
+        }
+      }
+    }
+
+    rerender(<App />);
+
+    // suspended render
+    await Profiler.takeRender();
+
+    {
+      // with a timeout > maxDelay, this passes
+      const { snapshot } = await Profiler.takeRender({
+        timeout: maxDelay + 100,
+      });
+
+      expect(snapshot.result?.data).toEqual({
+        viewer: {
+          __typename: "User",
+          age: 42,
+          id: "1",
+          name: "Jane Doe",
+          book: {
+            __typename: "TextBook",
+            id: "1",
+            publishedAt: "2024-01-01",
+            title: "The Book",
+          },
+        },
+      });
+    }
+
+    unmount();
+  });
+
+  it("should call invariant.error if min delay is greater than max delay", async () => {
+    using _consoleSpy = spyOnConsole.takeSnapshots("error");
+    const Profiler = createDefaultProfiler<ViewerQueryData>();
+
+    using _fetch = createSchemaFetch(schema, {
+      delay: { min: 3000, max: 1000 },
+    });
```

Comment:
Taking the above suggestions would change this expectation to something like this:

```suggestion
    expect(() => 
      createSchemaFetch(schema, {
        delay: { min: 3000, max: 1000 },
      })
    ).toThrowError(new InvariantError('...'))
```

Reply:
Calling `createSchemaFetch` itself won't throw, but I'm assuming you meant directly calling the `mock` function it returns. Unfortunately when I did something like this

```ts
    expect(() => {
        using schemaFetch = createSchemaFetch(schema, {
          delay: { min: 3000, max: 1000 },
        })
        schemaFetch.mock();
      }
    ).toThrowError(new InvariantError('...'))
```

I was getting the jest error `Jest worker encountered 4 child process exceptions, exceeding retry limit`, but I rewrote it as an integration style test here https://github.com/apollographql/apollo-client/pull/11774/commits/61b3be81a852517f65b7cb63fbd78be8d50a5a72

Reply:
Resolved in https://github.com/apollographql/apollo-client/pull/11774/commits/ba929d3a7cfc556cf533abe87d192784045c6c69